File: bigdata/sparkGmall/sparkmall-mock/src/main/scala/com/shangbaishuyao/utils/RandomDate.scala
package com.shangbaishuyao.utils

import java.util.Date

import scala.util.Random

/**
 * Desc: 随机日期工具类 <br/>
 * create by shangbaishuyao on 2021/3/9
 * @Author: 上白书妖
 * @Date: 14:37 2021/3/9
 */
object RandomDate {
  def apply(startDate: Date, endDate: Date, step: Int): RandomDate = {
    val randomDate = new RandomDate()
    val avgStepTime: Long = (endDate.getTime - startDate.getTime) / step
    randomDate.maxTimeStep = avgStepTime * 2
    randomDate.lastDateTime = startDate.getTime
    randomDate
  }

  class RandomDate {
    var lastDateTime = 0L
    var maxTimeStep = 0L

    def getRandomDate: Date = {
      val timeStep: Int = new Random().nextInt(maxTimeStep.toInt)
      lastDateTime = lastDateTime + timeStep

      new Date(lastDateTime)
    }
  }
}

File: bigdata/Gmall/Gmall-mockery/src/main/java/com/shangbaishuyao/utils/RandomDate.java
package com.shangbaishuyao.utils;

import java.util.Date;
import java.util.Random;
/**
 * Desc: 随机日期 <br/>
 * create by shangbaishuyao on 2021/3/13
 * @Author: 上白书妖
 * @Date: 19:23 2021/3/13
 */
public class RandomDate {

    Long logDateTime =0L;
    int maxTimeStep=0 ;

    public RandomDate (Date startDate , Date  endDate,int num) {
        Long avgStepTime = (endDate.getTime()- startDate.getTime())/num;
        this.maxTimeStep=avgStepTime.intValue()*2;
        this.logDateTime=startDate.getTime();
    }

    public  Date  getRandomDate() {
        int  timeStep = new Random().nextInt(maxTimeStep);
        logDateTime = logDateTime+timeStep;
        return new Date( logDateTime);
    }
}